include/sir/config.h

```diff
@@ -216,91 +257,137 @@
-# define SIRL_S_DEBUG  "debg"
+# if !defined(SIRL_S_DEBUG)
+#  define SIRL_S_DEBUG  "debg"
+# endif
 
 /** The maximum number of log files that may be registered at one time. */
-# define SIR_MAXFILES 16
+# if !defined(SIR_MAXFILES)
+#  define SIR_MAXFILES 16
+# endif
 
 /** The maximum number of plugin modules that may be loaded at one time. */
-# define SIR_MAXPLUGINS 16
+# if !defined(SIR_MAXPLUGINS)
+#  define SIR_MAXPLUGINS 16
+# endif
 
 /** The size, in characters, of the buffer used to hold file header format strings. */
-# define SIR_MAXFHEADER 128
+# if !defined(SIR_MAXFHEADER)
+#  define SIR_MAXFHEADER 128
+# endif
 
 /**
  * The maximum number of characters allowable in one log message. This
  * does not include accompanying formatted output (see ::SIR_MAXOUTPUT).
  */
-# define SIR_MAXMESSAGE 4096
+# if !defined(SIR_MAXMESSAGE)
+#  define SIR_MAXMESSAGE 4096
+# endif
 
 /** The size, in characters, of the buffer used to hold time format strings. */
-# define SIR_MAXTIME 64
+# if !defined(SIR_MAXTIME)
+#  define SIR_MAXTIME 64
+# endif
 
 /** The size, in characters, of the buffer used to hold millisecond strings. */
-# define SIR_MAXMSEC 5
+# if !defined(SIR_MAXMSEC)
+#  define SIR_MAXMSEC 5
+# endif
 
 /** The size, in characters, of the buffer used to hold level format strings. */
-# define SIR_MAXLEVEL 7
+# if !defined(SIR_MAXLEVEL)
+#  define SIR_MAXLEVEL 7
+# endif
 
 /**
  * The size, in characters, of the buffer used to hold process/appname
  * format strings.
  */
-# define SIR_MAXNAME 32
+# if !defined(SIR_MAXNAME)
+#  define SIR_MAXNAME 32
+# endif
 
 /**
  * The size, in characters, of the buffer used to hold system logger identity
  * strings.
  */
-# define SIR_MAX_SYSLOG_ID 128
+# if !defined(SIR_MAX_SYSLOG_ID)
+#  define SIR_MAX_SYSLOG_ID 128
+# endif
 
 /**
  * The size, in characters, of the buffer used to hold system logger category
  * strings.
  */
-# define SIR_MAX_SYSLOG_CAT 64
+# if !defined(SIR_MAX_SYSLOG_CAT)
+#  define SIR_MAX_SYSLOG_CAT 64
+# endif
 
 /** The maximum number of whitespace and miscellaneous characters included in output. */
-# define SIR_MAXMISC 7
+# if !defined(SIR_MAXMISC)
+#  define SIR_MAXMISC 7
+# endif
 
 /**
  * The size, in characters, of the buffer used to hold a sequence of styling
  * data in any color mode (the largest possible sequence, which is:
  * `\x1b[a;fb;m;rrr;ggg;bbb;fb;m;rrr;ggg;bbbm`) plus a null terminator.
  */
-# define SIR_MAXSTYLE 43
+# if !defined(SIR_MAXSTYLE)
+#  define SIR_MAXSTYLE 43
+# endif
 
 /** The maximum size, in characters, of final formatted output. */
-# define SIR_MAXOUTPUT \
-    (SIR_MAXMESSAGE + (SIR_MAXSTYLE * 2) + SIR_MAXTIME + SIR_MAXLEVEL + \
-        SIR_MAXNAME + (SIR_MAXPID   * 2) + SIR_MAXMISC + 1)
+# if !defined(SIR_MAXOUTPUT)
+#  define SIR_MAXOUTPUT \
+     (SIR_MAXMESSAGE + (SIR_MAXSTYLE * 2) + SIR_MAXTIME + SIR_MAXLEVEL + \
+         SIR_MAXNAME + (SIR_MAXPID   * 2) + SIR_MAXMISC + 1)
```

Comment:
Let's add a + 2 at the end for CRLF just to be safe. I don't remember how much breathing room MAXMISC gives us (there is some whitespace hard-coded)

Reply:
KK - can't hurt.